okhttp/src/nonJvmMain/kotlin/okhttp3/internal/-HttpUrlNonJvm.kt

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal
+
+import okhttp3.internal.CommonHttpUrl.isPercentEncoded
+import okhttp3.internal.NonJvmHttpUrl.writeCanonicalized
+import okio.Buffer
+
+internal object NonJvmHttpUrl {
```

Comment:
Reviewer request: any meaningful changes to any of these big functions? I’d prefer to skip a thorough review for anything that’s just moved around

Reply:
No.

I'll refactor in a follow up that will try to make the same common implementation for all of these including Charset as you suggest.  But would like to land this so I have a working implementation to start with.